writer.go

```diff
@@ -134,7 +154,7 @@ func (w *Writer) writeAndRetry(p Priority, s string) (int, error) {
 	w.Lock()
 	defer w.Unlock()
 
-	if w.conn != nil {
+	if w.conn != nil && !w.needReconnect {
```

Comment:
If the socket read is blocking, it doesn't seem like we would want to attempt it before every write.